integration-tests/src/tests/client/features/wallet_contract.rs

```diff
@@ -214,126 +230,258 @@ fn test_transaction_from_eth_implicit_account_fail() {
-
-    // Although ETH-implicit account can be zero-balance, we pick 1 here in order to make transfer later from this account.
-    let transfer_amount = 1u128;
-    let actions = vec![Action::Transfer(TransferAction { deposit: transfer_amount })];
-
-    if create_account {
-        // Create ETH-implicit account by funding it.
-        node.user()
-            .meta_tx(sender.clone(), eth_implicit_account.clone(), relayer.clone(), actions)
-            .unwrap()
-            .assert_success();
-    }
 
-    let target = carol_account();
-    let initial_balance = node.view_balance(&target).expect("failed looking up balance");
-
-    // TODO(eth-implicit) Append appropriate values to the RLP stream when proper `Wallet Contract` is implemented.
-    let mut stream = RlpStream::new_list(3);
-    stream.append(&target.as_str());
-    // The RLP trait `Encodable` is not implemented for `u128`. We must encode it as bytes.
-    // TODO(eth-implicit) Do not try to encode `u128` values directly, see https://github.com/near/nearcore/pull/10269#discussion_r1425585051.
-    stream.append(&transfer_amount.to_be_bytes().as_slice());
-    if authorized {
-        stream.append(&public_key.key_data());
-    } else {
-        stream.append(&other_public_key.key_data());
-    }
-    let rlp_encoded_data = stream.out().to_vec();
+    // Create ETH-implicit account by funding it.
+    // Although ETH-implicit account can be zero-balance, we pick a non-zero amount
+    // here in order to make transfer later from this account.
+    let deposit_for_account_creation = NEAR_BASE;
+    let actions = vec![Action::Transfer(TransferAction { deposit: deposit_for_account_creation })];
+    let nonce =
+        view_nonce(&env, relayer_signer.account_id, relayer_signer.signer.public_key.clone()) + 1;
+    let block_hash = *genesis_block.hash();
+    let signed_transaction = SignedTransaction::from_actions(
+        nonce,
+        relayer.clone(),
+        eth_implicit_account.clone(),
+        &relayer_signer.signer,
+        actions,
+        block_hash,
+    );
+    height = check_tx_processing(&mut env, signed_transaction, height, blocks_number);
 
-    let args = serde_json::json!({
-        "target": target.to_string(),
-        "rlp_transaction": rlp_encoded_data,
-    })
-    .to_string()
+    // The relayer adds its key to the eth implicit account so that
+    // can sign Near transactions for the user.
+    let relayer_pk = relayer_signer.signer.public_key.clone();
+    let action = Action::AddKey(Box::new(AddKeyAction {
+        public_key: relayer_pk,
+        access_key: AccessKey {
+            nonce: 0,
+            permission: AccessKeyPermission::FunctionCall(FunctionCallPermission {
+                allowance: None,
+                receiver_id: eth_implicit_account.to_string(),
+                method_names: vec!["rlp_execute".into()],
+            }),
+        },
+    }));
```

Comment:
Anyone can add access key to ETH-implicit account?

Reply:
Ahh, I see now. I mislooked that it signed by secret_key for ETH-implicit address.